engine/battlehack20/engine/container/code_container.py

```diff
@@ -55,7 +56,39 @@ def to_file(self, filename):
     @classmethod
     def from_file(cls, filename):
         with open(filename, 'rb') as f:
-            return cls.from_bytes(f.read())
+            return cls.from_bytes(cls.preprocess(f.read()))
+
+    @classmethod
+    def preprocess(cls, content):
+        """
+        Strips battlehack20.stubs imports from the code.
+
+        It removes lines containing one of the following imports:
+        - from battlehack20.stubs import *
+        - from battlehack20.stubs import a, b, c
+
+        The regular expression that is used also supports non-standard whitespace styles like the following:
+        - from battlehack20.stubs import a,b,c
+        - from  battlehack20.stubs  import  a,  b,  c
+
+        Go to https://regex101.com/r/bhAqFE/5 to test the regular expression with custom input.
+        """
+
+        pattern = r'^([ \t]*)from([ \t]+)battlehack20\.stubs([ \t]+)import([ \t]+)(\*|([a-zA-Z_]+([ \t]*),?([ \t]*))+)([ \t]*)$'
```

Comment:
Not important, but I think the following regex more closely matches what we want (the current regex allows `from battlehack20 import a,` which is not valid Python):

```
^([ \t]*)from([ \t]+)battlehack20\.stubs([ \t]+)import([ \t]+)(\*|([a-zA-Z_]+([ \t]*),([ \t]*))*[a-zA-Z_])([ \t]*)$
```

Reply:
Fixed, it now uses the regex you suggested.